src/include/migraphx/op/unique.hpp

```diff
@@ -0,0 +1,320 @@
+
+            // only axis = 0 is supported:
+            max_uniq_ct /= lens_x[0];
+            d_out.push_back({1, max_uniq_ct});
+            for(size_t idx = 1; idx < dim_x; idx++)
+                d_out.push_back({lens_x[idx], lens_x[idx]});
+        }
+        else
+        {
+            d_out.push_back({1, max_uniq_ct});
+        }
+
+        shape sh_y = {sh_x.type(), d_out};
+        // The three outputted Indices are just 1-D:
+        shape sh_idx{shape::int64_type, {d_out[0]}};
+
+        return {{sh_y, sh_idx, sh_idx, sh_idx}};
+    }
+
+    argument compute(const dyn_output& dyn_out, std::vector<argument> args) const
+    {
+        auto sh_x          = args.front().get_shape();
+        size_t dim_x       = sh_x.ndim();
+        auto lens_x        = sh_x.lens();
+        shape output_shape = dyn_out.computed_shape;
+        auto vec_ss        = output_shape.sub_shapes();
+        auto ct_x          = sh_x.elements();
+        shape sh_y         = {vec_ss[0].type(), {ct_x}};
+        shape sh_idx       = {vec_ss[1].type(), {ct_x}};
+        shape sh_x_idx     = {vec_ss[1].type(), {ct_x}};
+
+        argument res_y{sh_y};
+        argument res_y_idx{sh_idx};
+        argument res_x_rev_idx{sh_idx};
+        argument res_y_ct_idx{sh_idx};
+
+        std::vector<size_t> out_y_idx;
+        std::vector<size_t> out_x_rev_idx;
+        std::vector<size_t> out_y_ct;
+
+        // If axis is not none, for >1D tensors, we have to consider
+        // then, the uniqueness of chunks of sub-tensors: a subsequence of built-ins..
+        // For a built-in type, chunk_sz is of course = 1
+        size_t chunk_sz = 1;
+        if(axis)
+            chunk_sz = ct_x / lens_x[0]; // axis = 0 is supported.
+
+        visit_all(args.front(), res_y)([&](auto x, auto y_flat) {
+            using o_type = typename decltype(x)::value_type;
+            std::vector<o_type> x_in(x.begin(), x.end());
+
+            auto [y_indices, x_rev_indices, y_count] = sorted
+                                                           ? sorted_uniq_indices(x_in, chunk_sz)
+                                                           : unsorted_uniq_indices(x_in, chunk_sz);
+            const auto uniq_ct                       = y_indices.size();
+
+            // construct y from x[indices] in flattened form
+            // later we reshape y to the final shape..
+            auto y_dst = y_flat.begin();
+            for(size_t idx = 0; idx < uniq_ct; idx++)
+                y_dst = copy_n(x_in.begin() + y_indices[idx] * chunk_sz, chunk_sz, y_dst);
+
+            out_y_idx     = std::move(y_indices);
+            out_x_rev_idx = std::move(x_rev_indices);
+            out_y_ct      = std::move(y_count);
+
+            std::vector<size_t> y_lens = {uniq_ct};
+
+            // if axis is specified:
+            // the output shape has the n-1 dimensions of x
+            for(size_t idx = 1; axis && idx < dim_x; idx++)
+                y_lens.push_back(lens_x[idx]);
```

Comment:
This can be written using `std::copy`: `std::copy(lens_x.begin()+1, lens_x.begin()+dims_x, std::back_inserter(y_lens))`.